### DISCUSSION TITLE
Running diesel print-schema references a type of Tsvector which results in a compiler error.

### DISCUSSION THREAD

**Discussion by rdifrango** [diesel-rs/diesel#3042] [Q&A] [answered]
Running `diesel print-schema` references a type of Tsvector which results in a compiler error.  In my case I have a column of

```
error[E0412]: cannot find type `Tsvector` in this scope
```

If I change the type from:

```
text_search_vectors -> Nullable<Tsvector>
```

to

```
text_search_vectors -> Nullable<diesel_full_text_search::TsVector>
```

then it compiles fine, so not sure if this is a bug or something I did wrong when using `print-schema`?

For completeness the actual schema that I'm reverse engineering has:

```
text_search_vectors tsvector,
```

**Comment by weiznich** [comment, marked as answer]
This is currently expected behaviour as it's not possible provide out of the box support for a unbounded number of types. Therefore diesel itself provides only a core set of important types and leaves support for anything else to third party crates. The `diesel print-schema` tool does not know about which types are supported and which not, so you need to provide a suitable type definition if that's not done by diesel itself. Checkout [the cli guide](http://diesel.rs/guides/configuring-diesel-cli.html) for options to achieve this without manually modifying the generated schema.rs file.

**Reply by rdifrango** [reply]
@weiznich ok, so I followed the documentation and it added:

```
use diesel_full_text_search::types::*;
```

which per the compiler is

```
error[E0603]: module `types` is private
```

so there's a "bug" in the documenation.

the second thing is that I still think there is a bug in `print-schema` as it outputs as follows:

```
text_search_vectors -> Nullable<Tsvector>,
```

when it should be:

```
text_search_vectors -> Nullable<TsVector>,
```

Noticed the subtle difference of an uppercase `V` in the second one.

I could still be off mind you.

**Reply by weiznich** [reply]
> so there's a "bug" in the documenation.

Yes that sounds like the documentation is not correct.

> the second thing is that I still think there is a bug in print-schema as it outputs as follows:

That's a bug in diesel_full_text_search, not in `diesel print-schema`. See https://github.com/diesel-rs/diesel_full_text_search/issues/8 or https://github.com/diesel-rs/diesel_full_text_search/issues/26 for details.
